algosdk/abi.py

```diff
@@ -0,0 +1,570 @@
+from enum import IntEnum
+import re
+from unittest.mock import Base
+from . import error
+
+
+class BaseType(IntEnum):
+    Uint = 0
+    Byte = 1
+    Ufixed = 2
+    Bool = 3
+    ArrayStatic = 4
+    Address = 5
+    ArrayDynamic = 6
+    String = 7
+    Tuple = 8
+
+
+class Type:
+    """
+    Represents an ABI Type for encoding.
+
+    Args:
+        type_id (BaseType): type of ABI argument, as defined by the BaseType class above.
+        child_type (Type, optional): the type of the child_types array.
+        child_types (list, optional): list of types of the children for a tuple.
+        bit_size (int, optional): size of a uint/ufixed type, e.g. for a uint8, the bit_size is 8.
+        precision (int, optional): number of precision for a ufixed type.
+        static_length (int, optional): index of the asset
+
+    Attributes:
+        type_id (BaseType)
+        child_type (Type)
+        child_types (list)
+        bit_size (int)
+        precision (int)
+        static_length (int)
+    """
+
+    def __init__(
+        self,
+        type_id,
+        child_type=None,
+        child_types=list(),
```

Comment:
Not sure if it is the best practice of passing list() to default argument, but if it were me, I would do something like set it to be None as default, and check in constructor that: if `child_type` is None, then set it to be a list()

Reply:
I suggest the `None` as well since passing a list as a default argument could have astonishing effects :) 

https://stackoverflow.com/questions/1132941/least-astonishment-and-the-mutable-default-argument

Reply:
Thanks! Will fix this